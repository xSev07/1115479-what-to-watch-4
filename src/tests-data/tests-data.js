export const movies = [
  {
    id: `Hotel-2000`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`
    ],
    duration: 60,
    genre: `Drama`,
    inList: false,
    producer: `Wes Andreson`,
    rating: 8.9,
    reviews: [],
    title: `The Grand Budapest Hotel`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    votes: 240,
    year: 2000,
  },
  {
    id: `Bohemian-2018`,
    actors: [
      `Rami Malek`,
      `Lucy Boynton`,
      `Gwilym Lee`
    ],
    description: [
      `The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Музыкальный фестиваль Live Aid (1985)`
    ],
    duration: 134,
    genre: `Drama`,
    inList: false,
    producer: `Bryan Singer`,
    rating: 8.0,
    reviews: [],
    title: `Bohemian Rhapsody`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    votes: 220,
    year: 2018,
  },
  {
    id: `Macbeth-2015`,
    actors: [
      `Michael Fassbender`,
      `Marion Cotillard`,
      `Paddy Considine`
    ],
    description: [
      `From the Academy Award winning producers of The King’s Speech and acclaimed director Justin Kurzel, comes a visceral and visually breath-taking retelling of the classic tale about an ambitious Scottish lord who seizes the throne with the help of his wife. Starring Academy Award nominee Michael Fassbender and Academy Award winner Marion Cotillard, Macbeth is both visually epic and a breathtaking experience.`
    ],
    duration: 113,
    genre: `Documental`,
    inList: false,
    producer: `Justin Kurzel`,
    rating: 6.6,
    reviews: [],
    title: `Macbeth`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    votes: 220,
    year: 2015,
  },
  {
    id: `Aviator-2005`,
    actors: [
      `Leonardo DiCaprio`,
      `Cate Blanchett`,
      `Kate Beckinsale`
    ],
    description: [
      `The Aviator is a 2004 American epic biographical drama film directed by Martin Scorsese and written by John Logan. It stars Leonardo DiCaprio as Howard Hughes, Cate Blanchett as Katharine Hepburn, and Kate Beckinsale as Ava Gardner. The supporting cast features Ian Holm, John C. Reilly, Alec Baldwin, Jude Law as Errol Flynn, Gwen Stefani as Jean Harlow, Kelli Garner as Faith Domergue, Matt Ross, Willem Dafoe, Alan Alda, and Edward Herrmann.`,
      `Based on the 1993 non-fiction book Howard Hughes: The Secret Life by Charles Higham, the film depicts the life of Howard Hughes, an aviation pioneer and director of Hell's Angels. The film portrays his life from 1927–1947 during which time Hughes became a successful film producer and an aviation magnate while simultaneously growing more unstable due to severe obsessive–compulsive disorder (OCD).`
    ],
    duration: 170,
    genre: `Thriller`,
    inList: false,
    producer: `Martin Scorsese`,
    rating: 7.5,
    reviews: [],
    title: `Aviator`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    votes: 220,
    year: 2005,
  },
];

export const promoMovie = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014
};
