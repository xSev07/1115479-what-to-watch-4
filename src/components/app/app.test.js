import React from "react";
import renderer from "react-test-renderer";
import App from "./app";
import {movies, promoMovie} from "../../tests-data/tests-data";

it(`Should App render correctly`, () => {
  const tree = renderer
    .create(<App
      promoMovie={promoMovie}
      allMovies={movies}
      onMovieTitleClick={() => {}}
    />);

  expect(tree).toMatchSnapshot();
});
