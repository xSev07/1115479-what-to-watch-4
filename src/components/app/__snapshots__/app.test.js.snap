// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should App render correctly should render correctly basic 1`] = `
Array [
  <section
    className="movie-card"
  >
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <a
            href="/my-list"
            onClick={[Function]}
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </a>
        </div>
      </div>
    </header>
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/video/Hotel-2014"
              onClick={[Function]}
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Documental
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Thriller
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/Hotel-2000"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="175"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              The Grand Budapest Hotel
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/Bohemian-2018"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="175"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Bohemian Rhapsody
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/Macbeth-2015"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="175"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Macbeth
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/Aviator-2005"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="175"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Aviator
            </h3>
          </a>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Should App render correctly should render correctly loading error screen 1`] = `
<div
  style={
    Object {
      "background": "linear-gradient(-180deg,#180202 0%,#000 100%)",
    }
  }
>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/my-list"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </a>
      </div>
    </div>
  </header>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "height": "calc(100vh - 94px - 138px)",
        "justifyContent": "center",
        "width": "100vw",
      }
    }
  >
    <h2
      style={
        Object {
          "color": "#d9cd8d",
        }
      }
    >
      Извините, у нашего сервера лапки. Попробуйте позднее
    </h2>
    <svg
      enableBackground="new 0 0 512 512"
      height="125"
      viewBox="0 0 512 512"
      width="100"
      x="0"
      xmlns="http://www.w3.org/2000/svg"
      y="0"
    >
      <path
        d="m375.1,432.2c-14.7,0-28.8-4.5-41.8-13.2-3.4-2.3-14.4-5.7-22.9,0-29.3,19.9-66.1,16.9-93-7.4-7.8-7-19.5-7-27.3,0-15,13.5-32.7,20.7-51.2,20.7-47.4,0-86-46.5-86-103.7 0-41.6 20.6-79 52.5-95.4 8.2-4.2 12.5-13.3 10.7-22.3-1.8-8.7-2.7-17-2.7-25.2 0-57.2 38.6-103.7 86-103.7 32.1,0 60.8,21.7 76.3,56.1 6,13.5 23.9,12.5 29,9.6 11.8-6.8 24.3-10.3 37.1-10.3 46.3,0 84.9,45.5 85.9,101.4 0.1,4.9 2,9.7 5.3,13.3 17.9,19.7 28.2,47.6 28.2,76.6-5.68434e-14,56.9-38.6,103.5-86.1,103.5zm93.1-202c-4.8-75.7-59.1-133.9-126.4-133.9-13.1,0-25.9,2.3-38.2,6.7-23.6-38.7-62-62-104.3-62-70,0-126.9,64.9-126.9,144.6 0,6.2 0.4,12.5 1.1,18.9-37.7,25.8-61.5,72.9-61.5,123.9 0,79.8 56.9,144.6 126.9,144.6 23.2,0 45.3-7 64.9-20.4 19.6,13.4 41.7,20.4 64.9,20.4 18.4,0 36.6-4.6 53.2-13.4 16.7,8.8 34.8,13.4 53.2,13.4 70,0 126.9-64.9 126.9-144.6 0-36.8-12-71.4-33.8-98.2z"
        fill="#d9cd8d"
      />
      <path
        d="m285.9,288.9l56.8-56.8c8-8 8-20.9 0-28.9-8-8-20.9-8-28.9,0l-56.8,56.8-56.8-56.8c-8-8-20.9-8-28.9,0-8,8-8,20.9 0,28.9l56.8,56.8-56.8,56.8c-8,8-8,20.9 0,28.9 8,8 20.9,8 28.9,0l56.8-56.8 56.8,56.8c8,8 20.9,8 28.9,0 8-8 8-20.9 0-28.9l-56.8-56.8z"
        fill="#d9cd8d"
      />
    </svg>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Should App render correctly should render correctly loading screen (loading movies) 1`] = `
<div
  style={
    Object {
      "background": "linear-gradient(-180deg,#180202 0%,#000 100%)",
    }
  }
>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/my-list"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </a>
      </div>
    </div>
  </header>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "calc(100vh - 94px - 138px)",
        "justifyContent": "center",
        "width": "100vw",
      }
    }
  >
    <svg
      display="block"
      height="200"
      preserveAspectRatio="xMidYMid"
      style={
        Object {
          "margin": "auto",
        }
      }
      viewBox="0 0 100 100"
      width="200"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M85.141 70.133a40 40 0 00-71.194-36.487A42 40-62.865 0185.14 70.133"
        fill="#d9cd8d"
      >
        <animateTransform
          attributeName="transform"
          dur="1s"
          keyTimes="0;1"
          repeatCount="indefinite"
          type="rotate"
          values="0 50 51;360 50 51"
        />
      </path>
    </svg>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Should App render correctly should render correctly loading screen (loading promo) 1`] = `
<div
  style={
    Object {
      "background": "linear-gradient(-180deg,#180202 0%,#000 100%)",
    }
  }
>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/my-list"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </a>
      </div>
    </div>
  </header>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "calc(100vh - 94px - 138px)",
        "justifyContent": "center",
        "width": "100vw",
      }
    }
  >
    <svg
      display="block"
      height="200"
      preserveAspectRatio="xMidYMid"
      style={
        Object {
          "margin": "auto",
        }
      }
      viewBox="0 0 100 100"
      width="200"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M85.141 70.133a40 40 0 00-71.194-36.487A42 40-62.865 0185.14 70.133"
        fill="#d9cd8d"
      >
        <animateTransform
          attributeName="transform"
          dur="1s"
          keyTimes="0;1"
          repeatCount="indefinite"
          type="rotate"
          values="0 50 51;360 50 51"
        />
      </path>
    </svg>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
